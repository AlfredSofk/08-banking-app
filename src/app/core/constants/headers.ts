export const headers = {
    dispositivo: "PC",
    idioma: "es",
    uuid: "02e3eb27-6fb1-e542-e157-c301cc77ad2c",
    ip: "localhost",
    horaTransaccion: "string",
    llaveSimetrica: "xaqVyedHolrJB9vW4lIj5u9nuWIiaPpAQoOK4hm2j+Q=",
    vectorInicializacion: "Gz3MLPvKU1T5Pc3FfmNYPe9nuWIiaPpAQoOK4hm2j+Q="

}