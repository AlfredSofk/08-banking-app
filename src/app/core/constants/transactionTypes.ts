export enum TransactionTypes {
    WITHDRAW = 'withdraw',
    DEPOSIT = 'deposit',
    DEPOSIT_ACCOUNT = 'deposit-account',
    TRANSFER = 'transfer',
    PURCHASE_WEB = 'purchase-web',
    PURCHASE_STORE = 'purchase-store',
}