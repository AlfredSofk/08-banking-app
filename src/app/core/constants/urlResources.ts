import { environment } from "../../environment/environment";

export const urlResources = {
    generateToken: `${environment.apiBaseUrl}/utils/generate`,
    getClientData : `${environment.apiBaseUrl}/client/findByUserName`,
    getTransaction: (transactionType: string) => `${environment}/transactions/${transactionType}`,
    
};