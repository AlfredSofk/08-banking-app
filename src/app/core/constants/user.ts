export enum USER_LOGIN {
    username = "pablo",
    password = "12qwaszx"
}