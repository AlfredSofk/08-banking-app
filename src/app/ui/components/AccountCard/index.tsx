

import { decryptAES } from '../../../core/utils/decoder'
import './style.scss'

interface Props {
    accountNumber : string,
    balance: number
}

export const AccountCard = ({accountNumber, balance} : Props) => {
    return (
        <div className="account-card">
            {/* <h3 className="account-card__title">{accountName}</h3> */}
            <p className="account-card__number" data-testid="accountNumber">NÃºmero de Cuenta: {accountNumber}</p>
            <p className="account-card__balance" data-testid="balance">Saldo: ${balance.toFixed(2)}</p>
        </div>
    )
}