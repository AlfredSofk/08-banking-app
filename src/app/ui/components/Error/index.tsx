import React from "react";
import "./style.scss";

interface ErrorMessageProps {
  title?: string; // Título del mensaje de error
  description?: string; // Descripción del error
  errorDetails?: string; // Detalles adicionales del error
}

export const ErrorScreen: React.FC<ErrorMessageProps> = ({
  title = "Something went wrong",
  description = "An unexpected error occurred. Please try again later.",
  errorDetails,
}) => {
  return (
    <section
      className="error-message"
      role="alert"
      aria-live="assertive"
      aria-atomic="true"
    >
      <header className="error-message__header">
        <h1 className="error-message__title">{title}</h1>
      </header>
      <div className="error-message__body">
        <p className="error-message__description">{description}</p>
        {errorDetails && (
          <details className="error-message__details">
            <summary className="error-message__summary">More details</summary>
            <p>{errorDetails}</p>
          </details>
        )}
      </div>
    </section>
  );
};