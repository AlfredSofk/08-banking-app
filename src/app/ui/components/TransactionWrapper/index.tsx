import {
  TransactionTypes,
  TransactionTargets,
} from "../../../core/constants/transactionTypes";
import { Transaction } from "../Withdraw";
import { useTransactions } from "../../../core/hooks/useTransaction";

interface Props {
  transactionType: string;
  // children: ReactNode | ReactNode[];
}

export const TransactionWrapper = ({ transactionType }: Props) => {
  const {
    state,
    retiroCajeroATM,
    depositarCajeroATM,
    depositarAgencia,
    transferencias,
    compraWeb,
    compraEstablecimiento,
  } = useTransactions();
  console.log({ transactionType });
  return (
    <>
      {(() => {
        switch (transactionType) {
          case TransactionTypes.WITHDRAW:
            return (
              <Transaction
                titleTransaction="retiro ATM"
                state={state}
                transactionHook={retiroCajeroATM}
                transactionTarget={TransactionTargets.RETIRO}
              />
            );
          case TransactionTypes.DEPOSIT:
            return (
              <Transaction
                titleTransaction="deposito ATM"
                state={state}
                transactionHook={depositarCajeroATM}
                transactionTarget={TransactionTargets.DEPOSITO}
              />
            );
          case TransactionTypes.DEPOSIT_ACCOUNT:
            return (
              <Transaction
                titleTransaction="deposito Agencia"
                state={state}
                transactionHook={depositarAgencia}
                transactionTarget={TransactionTargets.DEPOSITO}
              />
            );
          case TransactionTypes.TRANSFER:
            return (
              <Transaction
                titleTransaction="Tranferencia"
                state={state}
                transactionHook={transferencias}
                transactionTarget={TransactionTargets.TRANSFERENCIA}
              />
            );
          case TransactionTypes.PURCHASE_WEB:
            return (
              <Transaction
                titleTransaction="Compra Web"
                state={state}
                transactionHook={compraWeb}
                transactionTarget={TransactionTargets.COMPRA}
              />
            );
          case TransactionTypes.PURCHASE_STORE:
            return (
              <Transaction
                titleTransaction="Compra Establecimiento"
                state={state}
                transactionHook={compraEstablecimiento}
                transactionTarget={TransactionTargets.COMPRA}
              />
            );
          default:
            return <div>No se encontró el tipo de transacción</div>;
        }
      })()}
    </>
  );
};
