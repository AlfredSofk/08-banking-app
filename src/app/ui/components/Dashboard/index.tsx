import { IAppState } from "../../../core/interfaces/bankAccount";
import { AccountCard } from "../AccountCard";
import { GeneralBalance } from "../GeneralBalance";
import './style.scss'

interface Props { 
    state : IAppState
}


export const Dashboard = ({state} : Props) => {

    const {userAccounts, saldoGlobal} = state

    return (
        <section className="accounts-overview" aria-labelledby="overview-heading">
        <h2 id="overview-heading" className="accounts-overview__title" role="heading">Resumen de Cuentas</h2>
        <GeneralBalance balance={saldoGlobal} />
        <div className="accounts-overview__accounts">
          {userAccounts.length === 0 && <p>No hay cuentas disponibles</p>}
          {userAccounts.map((account, index) => (
            <AccountCard
              key={index}
            //   accountName={get}
              accountNumber={account.number}
              balance={account.amount}
            />
          ))}
        </div>
      </section>
    )

}