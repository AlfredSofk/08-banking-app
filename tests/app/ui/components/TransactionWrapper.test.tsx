import React from 'react';
import { render, screen } from '@testing-library/react';
import { TransactionWrapper } from '../../../../src/app/ui/components/TransactionWrapper';
import { vi } from 'vitest';
import { TransactionTypes } from '../../../../src/app/core/constants/transactionTypes';

// Mock del custom hook useTransactions
vi.mock('../../../../src/app/core/hooks/useTransaction', () => ({
    useTransactions: vi.fn(),
}));


describe('test para el componente TransactionWrapper', async () => {

    const {useTransactions} = await import('../../../../src/app/core/hooks/useTransaction');
    const mockUseTransactions = {
        state: {},
        retiroCajeroATM: vi.fn(),
        depositarCajeroATM: vi.fn(),
        depositarAgencia: vi.fn(),
        transferencias: vi.fn(),
        compraWeb: vi.fn(),
        compraEstablecimiento: vi.fn(),
    };

    beforeEach(() => {
        vi.clearAllMocks();
        (useTransactions as unknown as ReturnType<typeof vi.fn>).mockReturnValue({...mockUseTransactions});
    });

    test('match snapshot', () => {
        const { asFragment } = render( 
        <TransactionWrapper />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test('renderiza el componente Transaction para retiro ATM', () => {
        render(<TransactionWrapper transactionType={TransactionTypes.WITHDRAW} />);

        // Verificar que se renderiza el componente de retiro ATM
        expect(screen.getByTestId('title-transaction')).toHaveTextContent('retiro ATM');
    });

    test('renderiza el componente Transaction para deposito ATM', () => {
        render(<TransactionWrapper transactionType={TransactionTypes.DEPOSIT} />);

        // Verificar que se renderiza el componente de deposito ATM
        expect(screen.getByTestId('title-transaction')).toHaveTextContent('deposito ATM');
    });

    test('renderiza el componente Transaction para deposito Agencia', () => {
        render(<TransactionWrapper transactionType={TransactionTypes.DEPOSIT_ACCOUNT} />);

        // Verificar que se renderiza el componente de deposito Agencia
        expect(screen.getByTestId('title-transaction')).toHaveTextContent('deposito Agencia');
    });

    test('renderiza el componente Transaction para transferencias', () => {
        render(<TransactionWrapper transactionType={TransactionTypes.TRANSFER} />);

        // Verificar que se renderiza el componente de transferencias
        expect(screen.getByTestId('title-transaction')).toHaveTextContent('Transferencia');
    });

    test('renderiza el componente Transaction para compra Web', () => {
        render(<TransactionWrapper transactionType={TransactionTypes.PURCHASE_WEB} />);

        // Verificar que se renderiza el componente de compra Web
        expect(screen.getByTestId('title-transaction')).toHaveTextContent('Compra Web');
    });

    test('renderiza el componente Transaction para compra Establecimiento', () => {
        render(<TransactionWrapper transactionType={TransactionTypes.PURCHASE_STORE} />);

        // Verificar que se renderiza el componente de compra Establecimiento
        expect(screen.getByTestId('title-transaction')).toHaveTextContent('Compra Establecimiento');
    });
});