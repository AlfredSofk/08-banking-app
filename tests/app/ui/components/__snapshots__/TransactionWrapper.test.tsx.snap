// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test para el componente TransactionWrapper > match snapshot 1`] = `
<DocumentFragment>
  <div>
    No se encontró el tipo de transacción
  </div>
</DocumentFragment>
`;
